lib/versioned/test.js

```diff
@@ -22,7 +22,7 @@ function Test(directory, pkgVersions, opts = {}) {
 
   this.name = dirname === 'versioned' ? pkg.name : dirname
   this.directory = directory
-  if (testPatterns) {
+  if (testPatterns && testPatterns.length) {
```

Comment:
would do a value check instead of truthy to be explicit `testPatterns.length > 0`